frontends/ast/simplify.cc

```diff
@@ -4936,6 +4849,9 @@ void AstNode::mem2reg_as_needed_pass1(dict<AstNode*, pool<std::string>> &mem2reg
 	{
 		AstNode *mem = id2ast;
 
+		if (integer < (unsigned)mem->unpacked_dimensions)
+			input_error("Insufficient number of array indices for %s.\n", log_id(str));
```

Comment:
This is really a shortcoming in Yosys, which currently treats anything with unpacked ranges as "memory", where the memory word consists of the packed part, and only indexing which ultimately selects the packed part is allowed (contrary to the LRM).

I'd rather not add tests for this shortcoming; I think it would be better to consider what could possibly be done about this in the future. Perhaps automatic `mem2reg` conversion of "memory" which has unpacked parts selected?